torchsde/_core/methods/milstein.py

```diff
@@ -60,7 +60,7 @@ def step(self, t0, t1, y0):
         g_prod_I_k = self.sde.g_prod(t0, y0, I_k)
 
         if self.options[METHOD_OPTIONS.grad_free]:
-            g = self.sde.g(t0, y0)
+            g = self.sde.g(t0, y0) if self.sde.noise_type is not NOISE_TYPES.scalar else self.sde.g(t0, y0)[:,:,0]
```

Comment:
Could you do something similar to [this](https://github.com/google-research/torchsde/blob/19062d5e7ce8a13b0e71092e2153c6f74a947943/torchsde/_core/methods/srk.py#L74)?

Reply:
Done